### ISSUE TITLE
rolling release

### ISSUE THREAD

**Issue by pinventado** [kevinwortman/tuffix#13]
I've had a couple of students ask if they can use last year's version of tuffix. I wonder how difficult it would be for us to support a rolling release instead of having to download a new VM image every year.

Would running an updated tuffixize script inside a VM solve this?

**Comment by kevinwortman** [comment]
Agreed that we should think about an upgrade path for students with an older VM. So far this has not been a supported use case so the tools do not support it at all.

Running a newer tuffixize on an older VM won't work because tuffixize expects to run on a specific Xubuntu release. Over time as Ubuntu has added packages to their repositories, we've been adding those packages to tuffixize and taking out compile-from-source and other manual processes.

"Someone" could re-engineer the script to be backwards- and forwards-compatible. This would be a good undergraduate capstone project.

FWIW, now that we only make one release per year, I don't think asking students to reinstall the VM is unreasonable. It takes about 5 minutes per year and a typical student will only upgrade 3-4 times in their life. :P

**Comment by pinventado** [comment]
True :) Let's keep this in the back burner until we find someone who wants to work on this :D
